### DISCUSSION TITLE
`AutoSuggest` component

### DISCUSSION THREAD

**Discussion by GerardasB** [iTwin/iTwinUI#2160] [Ideas]
### Feature

AppUI team is deprecating `AutoSuggest` from `@itwin/core-react` [storybook](https://itwin.github.io/appui/storybook/?path=/story/components-autosuggest--basic&globals=resizer:true)
The suggested replacement for the upcoming deprecation is to use iTwinUI components, but it would be great to provide an example as well.

`AutoSuggest` is similar to the existing iTwinUI component `ComboBox`. It's an input with a popover, that has keyboard navigation. However, the indented usage of a combo box is selecting an option from known options, while auto-suggest is about typing in anything and using the menu items as suggestions.
 
Alternatively, I've tried building the `AutoSuggest` with more basic components like `Input`/`Popover`/`DropdownMenu` focus&keyboard navigation gets tricky i.e. currently `Popover` by default focuses the floating element and there seems to be no good way to override that (auto-suggest input should stay focused when the popover is opened, which is analogous to `ComboBox` behavior).

### Examples

Here's my experimentation w/ iTwinUI `ComboBox` [stackblitz](https://stackblitz.com/edit/github-jrzgwy?file=src%2FApp.tsx)

**Comment by mayank99** [comment]
Converting this to a discussion, since this isn't ready for implementation yet. We might want to expose this as a prop in `ComboBox`. TBD.
